### PR TITLE
test: rounding tests low slashing factors

### PR DESCRIPTION
**Pull request by 8sunyuan**
**Motivation:**

Testing coverage for rounding edge cases.

**Modifications:**

Added test file in `Register_Allocate_Slash_VerifyWC_.t.sol`
The _init setup creates an Operator registered to an OperatorSet who gets slashed down to 1 maxMagnitude.

- Removed duplicate function `assert_Snap_StakeBecomeUnslashable`
- Added setter for BCSF on EPMWrapper
- Fixed `assert_Snap_DSF_State_WithdrawalAsShares()` for when fully slashed
- Fixed `assert_Snap_DSF_State_Delegation()` to use staker's operator magnitudes instead of staker's magnitude itself
- Additional checks helpers
- Added helper `getSlashingFactor` for User.t.sol

**Result:**
Integration Tests with 1 operator MaxMagnitude
`test_slashBC_startCompleteCP_queue_complete`
`test_slash_undelegate_redeposit`
`test_slash_undelegate_completeAsTokens_verifyWC`
`test_slash_undelegate_completeAsShares_startCompleteCP`
`test_queueAllShares_completeAsTokens`

Integration Tests with 1 BCSF
`test_verifyWC_startCompleteCP`
`test_slashFullyBC_deposit`
`test_slashAVS_delegate_startCompleteCP`

Integration Tests with High DSF and repeat deposits
`test_multiple_deposits`

### CODE REVIEW

File: src/test/integration/tests/eigenpod/Register_Allocate_Slash_VerifyWC_.t.sol [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+        slashParams.wadsToSlash[0] = WAD;
+        avs.slashOperator(slashParams);
+        check_Base_Slashing_State(operator, allocateParams, slashParams);
+
+        // 5. undelegate results in 0 delegated shares removed since operator has 0 magnitude and staker is fully slashed too
+        Withdrawal[] memory withdrawals = staker.undelegate();
+        bytes32[] memory withdrawalRoots = _getWithdrawalHashes(withdrawals);
+        check_Undelegate_State(staker, operator, withdrawals, withdrawalRoots, strategies, initDepositShares, 0.toArrayU256());
+
+        // 6. redeposit (start/complete checkpoint)
```
[the remark this answers is not in the document]
Comment (8sunyuan) [reply src/test/integration/tests/eigenpod/Register_Allocate_Slash_VerifyWC_.t.sol:140]: Good point, I personally define `redeposit` as adding shares so this can be either from verify wc of new validators,  completing checkpoints, or completing any queued withdrawals. 

I'll update this test to randomly choose between those 2 paths on the EigenPod

[the remark this answers is not in the document]
Comment (8sunyuan) [reply src/test/integration/tests/eigenpod/Register_Allocate_Slash_VerifyWC_.t.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/1178/commits/6f9b2d5e6fc11228a3810c30b8e2e4b1bfe2009d
